File: Stack/stack.py
class Stack:
    def __init__(self):
        self.items = []
    
    def push(self, item):
        self.items.append(item)
    
    def size(self):
        return len(self.items)
    
    def is_empty(self):
        return len(self.items) == 0
    
    def __call__(self):
        print(self.items)
    
    def pop(self):
        if not self.is_empty():
            return self.items.pop()
        else:
            print("Stack Underflow")

    def top(self):
        if not self.is_empty():
            return self.items[-1]
        else:
            print("Stack is empty")

stack = Stack()

stack.push(10)
stack.push(20)
stack.push(100)
stack.push(1000)

print(f'Size: {stack.size()}')
stack.pop()
print(f"Top element: {stack.top()}")
print(f"Is empty: {stack.is_empty()}")
print(f"STACK elements: {stack.items}")

File: stack.cpp
#include <iostream>
#include <stack>

using namespace std;

int main() {
    stack<int> myStack;

    // Push elements onto the stack
    myStack.push(10);
    myStack.push(20);
    myStack.push(30);

    // Print the top element
    cout << "Top element of stack: " << myStack.top() << endl;

    // Pop elements from the stack
    myStack.pop();
    myStack.pop();

    // Check if the stack is empty
    if (myStack.empty()) {
        cout << "Stack is empty" << endl;
    } else {
        cout << "Stack is not empty" << endl;
    }

    return 0;
}
